packages/db/src/provider.ts

```diff
@@ -0,0 +1,59 @@
+import omit from 'lodash/omit';
+import isNil from 'lodash/isNil';
+import isValidHostname from 'is-valid-hostname';
+
+import { ProviderRecord, PostgresProviderRecord } from '@cumulus/types';
+import { PostgresValidationError } from '@cumulus/errors';
+
+import KMS from '@cumulus/aws-client/KMS';
+
+export const encryptValueWithKMS = (
+  value: string
+): Promise<string> => {
+  if (process.env?.provider_kms_key_id === undefined) {
+    throw new Error('env variable provider_kms_key_id must be set');
+  }
+  return KMS.encrypt(process.env?.provider_kms_key_id, value);
+};
+
+export const postgresProviderFromCumulusProvider = async (
+  data: ProviderRecord,
+  encryptMethod: Function = encryptValueWithKMS
+) => {
+  let username: string | undefined;
+  let password: string | undefined;
+  if (data.username) {
+    username = await encryptMethod(data.username);
+  }
+  if (data.password) {
+    password = await encryptMethod(data.password);
+  }
+  return ({
+    ...(omit(data, ['id', 'encrypted', 'createdAt', 'updatedAt'])),
+    name: data.id,
+    created_at: data.createdAt,
+    updated_at: data.updatedAt,
+    username,
+    password,
+  });
+};
+export const nullifyUndefinedProviderValues = (
```

Comment:
Can you remind me what this is used for? I'm trying to recall

Reply:
I think it changes all `undefined` to `null` before writing to rds because there is no concept of `undefined`. There is only `true`, `false` and `null` as a `Boolean` type, I believe.

Reply:
See: http://knexjs.org/#Builder-update for more on why.    `undefined` keys are ignored on updates queries.